File: easy/2574_left_and_right_sum_differences/solution.py
from typing import List


class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        differences = []

        numsSum = sum(nums)
        leftSum = 0

        for num in nums:
            differences.append(abs(leftSum * 2 + num - numsSum))
            leftSum += num

        return differences

File: easy/2574_left_and_right_sum_differences/solution.cpp
#include <numeric>
#include <vector>

class Solution {
public:
  std::vector<int> leftRightDifference(std::vector<int> &nums) {
    std::vector<int> differences;

    const int numsSum = std::reduce(nums.begin(), nums.end());
    int leftSum = 0;

    for (const int &num : nums) {
      differences.push_back(abs(leftSum * 2 + num - numsSum));
      leftSum += num;
    }

    return differences;
  }
};
